src/components/TodoList/TodoList.tsx

```diff
@@ -1,210 +1,122 @@
-/* eslint-disable max-len */
-import React from 'react';
+import React, { useMemo } from 'react';
+import classNames from 'classnames';
+import { useAppDispatch, useAppSelector } from '../../app/hooks';
+import { TotoActions } from '../../features/currentTodo';
+import { Todo } from '../../types/Todo';
+import { SelectValue } from '../../types/SelectValues';
 
 export const TodoList: React.FC = () => {
+  const dispatch = useAppDispatch();
+  const todos = useAppSelector(state => state.todos);
+  const currentTodo = useAppSelector(state => state.currentTodo);
+  const { query, status } = useAppSelector(state => state.filter);
+  const setCurrentTodo = (todo: Todo) => dispatch(TotoActions.setTodo(todo));
+
+  const filterForStatusTodo = () => {
+    switch (status) {
+      case SelectValue.Completed:
+        return todos.filter((todo) => todo.completed);
+      case SelectValue.Active:
+        return todos.filter((todo) => !todo.completed);
+      default:
+        return todos;
+    }
+  };
+
+  const todosFilter = (todosFromServer: Todo[], search: string) => {
+    const queryLower = search.toLowerCase().trim();
+
+    if (queryLower === '') {
+      return todosFromServer;
+    }
+
+    return todosFromServer.filter((todo:Todo) => todo.title
+      .toLowerCase()
+      .includes(queryLower));
+  };
+
+  const visibleTodos = useMemo(() => {
+    return todosFilter(filterForStatusTodo(), query);
+  }, [todos, query, status]);
```

Comment:
Make sure that the dependencies array is correct. You should pass todosFilter, and filterForStatusTodo there as well. And remove todos and status. Wrap those functions in their own useMemo() and add dependencies there. Those functions are not computationally expensive, but it's still reasonable to use them as part of the memoization process, as it's part of the filtering process.